runtime/libc/wasi_sdk_backing.c

```diff
@@ -297,14 +299,26 @@ wasi_errno_t wasi_snapshot_preview1_clock_time_get(
  * Read environment variable data.
  * The sizes of the buffers should match that returned by `environ_sizes_get`.
  * 
- * @param environ_retptr
- * @param environ_buf_retptr
+ * @param environ_baseretptr
+ * @param environ_buf_baseretptr
+ * @return WASI_ESUCCESS
  */
 wasi_errno_t wasi_snapshot_preview1_environ_get(
-    wasi_size_t environ_retptr, 
-    wasi_size_t environ_buf_retptr
+    wasi_size_t environ_baseretptr, 
+    wasi_size_t environ_buf_baseretptr
 ) {
-    wasi_unsupported_syscall(__func__);
+    wasi_size_t *environ_offset_vector = (wasi_size_t *)get_memory_ptr_for_runtime(environ_baseretptr, runtime_environc);
+    char *environ_buf = get_memory_ptr_for_runtime(environ_buf_baseretptr, runtime_environv_buf_len);
+    extern char **environ;
```

Comment:
Why do we need an extern here?

Reply:
It's part of the API. From `environ.7`: "This variable must be declared in the user program, but is declared in the header file <unistd.h> if the _GNU_SOURCE feature test macro is defined."